src-generate/Generator.hs

```diff
@@ -461,6 +462,19 @@ getInsertFunc maxCapability typeDesc = unlines ["{-# INLINE " ++ funcName ++ " #
         toCasesHelper (_, x) [] = ["| otherwise " ++ x]
         toCasesHelper (c, x) xs = ("| " ++ c ++ " " ++ x) : xs
 
+-- | Generate a function to generate a vector from an index-function
+getGenerateFunc :: TypeDesc -> String
+getGenerateFunc typeDesc = unlines
+    ["{-# INLINE[1] " ++ funcName ++ " #-}"
```

Comment:
Fine! No hurry.
I'm really not firm with those unboxed functions; would what you mean look essentially like the unboxing in `mapVector`?

BTW kudos for the work this really nice package.